### ISSUE TITLE
loss function at surrogate.py

### ISSUE THREAD

**Issue by IsmailBData** [khalil-research/PyEPO#56]
Hello,

I hope you’re all doing great!

I have a question regarding the following line:

loss = -obj + 2 * torch.einsum("bi,bi->b", cp, w) - z.squeeze(dim=-1)

I couldn’t find its corresponding equation or explanation in the paper that discusses SPO.

Any help would be greatly appreciated.

Thank you in advance!
